mono/mini/mini-arm.c

```diff
@@ -5959,19 +5959,19 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 			bb->spill_slot_defs = g_slist_prepend_mempool (cfg->mempool, bb->spill_slot_defs, ins);
 			break;
 		case OP_GC_SAFE_POINT: {
-#if defined (USE_COOP_GC)
-			const char *polling_func = NULL;
-			guint8 *buf [1];
+			if (mono_threads_is_coop_enabled ()) {
```

Comment:
If we're filtering safepoint placing at construction time, I don't see the point of checking it again here. The same applies for other backends - maybe we want the s390x assert instead?